Sashanity/NoteTube
# Favorite, Unfavorite, and Check if a single note was favorited

Favorite Note

To test on Postman, follow these settings:

url: http://localhost:5000/notetube-f3f9c/us-central1/api/favoriteNote
type: POST

PARAMS:
-noteid: The ID of the note you want to preview
-token: The token you get when you logged in
-public: If the note is public or not

RETURN (if successful):
-Status: Should say "Successful" if the note was favorited

NOTES:
-An object will be created in the Favorites collection in Firestore

Unfavorite Note

To test on Postman, follow these settings:

url: http://localhost:5000/notetube-f3f9c/us-central1/api/unfavoriteNote
type: DELETE

PARAMS:
-noteid: The ID of the note you want to preview
-token: The token you get when you logged in
-public: If the note is public or not

RETURN (if successful):
-Status: Should say "Successful" if the note was unfavorited

NOTES:
-An object will be removed in the Favorites collection in Firestore

Has Favorited Note

To test on Postman, follow these settings:

url: http://localhost:5000/notetube-f3f9c/us-central1/api/hasFavoritedNote
type: Get

PARAMS:
-noteid: The ID of the note you want to preview
-token: The token you get when you logged in
-public: If the note is public or not

RETURN (if successful):
-Status: Should say "Successful" if the note was favorited
-Favorited: Will say whether the note has been favorited or not

EDIT: Added Favorites List API. Please review this one as well

Favorite List

To test on Postman, follow these settings:

url: http://localhost:5000/notetube-f3f9c/us-central1/api/favoriteList
type: GET

PARAMS:
-token: The token you get when you logged in

RETURN (if successful):
-Status: Should say "Successful" if the note was favorited
-List: If successful, will return the list of notes the user has favorited

EDIT2: Added public note param to each API. Favoriteslist still needs to be updated.
